### DISCUSSION TITLE
Is there way to hide module hook?

### DISCUSSION THREAD

**Discussion by PlatFlat** [mywalkb/LSPosed_mod#19] [Q&A]
I just noticed most apps (starbucks, revolut) is detecting LSPosed module hook. Example adding target app to an LSPosed module caused it to triggered some sort of crash.

**Comment by mywalkb** [comment]
Have you tried shamiko?
At this moment it is the only possible solution even if it is not perfect.
Is not easy hide injection in a process.
You can try to replace zygisk with Dr-TSNG/ZygiskOnKernelSU, it implement native bridge for injection, maybe this injection is not yet  detected.
Is not only for kernelsu but is also a magisk's module.

**Reply by PlatFlat** [reply]
Yup, tested Riru/Zygisk on Magisk with MomoHider/Shamiko. My current test case here is on Shamiko with ZygiskOnKernelSU on A13.

The detection only occurred when the app was selected in the LSPosed scope of any LSPosed module.
Once it is removed from any LSPosed module scope, it won't crash.

It seems like LSposed/module injection is causing it. Another example to test out is by installing Memory Detector by HuskyDG and Stillhard, then having any LSPosed module add Memory Detector as a scope. You will see two injection processes found

**Reply by mywalkb** [reply]
You have 2 ways: to develope a LSPosed module which hook the detection and bypass it also with native hook if is required, as was done [here](https://github.com/LSPosed/LSPosed/issues/471#issuecomment-817294383) or wait for shamiko to improve.

> caused it to triggered some sort of crash.
> It seems like LSposed/module injection is causing it

Crash or simply close the app without warning? Have you seen logcat?

Are there too many ways to detect injection so is very hard found a generic way to bypass.

**Reply by PlatFlat** [reply]
I think this is going to be a challenge for users, especially those who still care for privacy or uses XPrivacyLua,  I am also not sure if dev of shamiko will help with the module detection of [LSPosed, as indicated by them](https://github.com/LSPosed/LSPosed/issues/2686#issuecomment-1676574402). 

> Crash or simply close the app without warning? Have you seen logcat?

Generally it just close the affected app. On a recent version, for example, Revolut v9.25. Once it detected root or lsposed, it instantly rebooted the phone until you remove any root or hooks. I tested this and there's has been [similar reports](https://forum.xda-developers.com/t/discussion-magisk-the-age-of-zygisk.4393877/post-89011233)

**Reply by PlatFlat** [reply]
I saw this [article ](https://github.com/HuskyDG/huskydg.github.io/blob/master/blog/detect_magisk_xposed.md#xposed) by HuskyDG.
and I also heard that you can check xposed api classes through reflection. I think this is how apps like Revolut and Singpass gets force closed when they get hooked.

Could these Xposed Hooks be hidden in any way?

**Comment by etnperlong** [comment]
In my opinion, the hide provided by Shamiko is imperfect. Have you tried HSBC UK or any other HSBC apps from different countries?
I think the newer detection is deployed everywhere.

I use KernelSU 0.6.8 + ZygiskOnKernelSU v4-0.7.1 + Shamiko v0.7.3
with LSPosed v1.9.1 and HideMyAppList v3.2 blacklist enabled.

I cannot open HSBC UK from today, but I definitely can open it yesterday.
There is no update from Play Store but I think there is some hotfix update.

Btw, Revolut, Chase UK and Starbucks work well on my rooted phone for a longer time.

**Reply by etnperlong** [reply]
The abnormal environment can be detected by NativeTest from this Telegram channel
https://t.me/nullptr_dev/34

I still haven't found any way to fix it, so feel free to let me know if you have a solution. :)

**Reply by mywalkb** [reply]
> In my opinion, the hide provided by Shamiko is imperfect. Have you tried HSBC UK or any other HSBC apps from different countries? I think the newer detection is deployed everywhere.

Is cat & mouse game. For each update a new method is created (for bypass and for detect).
Some months ago I started this [project](https://github.com/mywalkb/DenylistUnmount) a similar module to shamiko but open source.
I tried to create a generic way to hide magisk and lsposed but I failed. 
Now even the libraries are obfuscated so it's difficult to do analysis.

**Reply by mywalkb** [reply]
> I still haven't found any way to fix it, so feel free to let me know if you have a solution. :)

Have you tried [this](https://github.com/HuskyDG/zygisk-maphide)? I do not have kernelsu so I can't test it.

**Reply by etnperlong** [reply]
No luck :( The HSBC apps are still crashing

**Reply by mywalkb** [reply]
> No luck :( The HSBC apps are still crashing

@etnperlong could you test this [action](https://github.com/mywalkb/LSPosed_mod/actions/runs/6591226695) ?
There is a new option in settings "Enable watchdog logs", disable it and test HSBC.

**Reply by mywalkb** [reply]
there is a conversation [here](https://github.com/LSPosed/LSPosed/pull/2822) for this new option
